src/charm.py

```diff
@@ -95,6 +98,38 @@ def __init__(self, *args):
             self.fiveg_n3_provider.on.fiveg_n3_request, self._on_fiveg_n3_request
         )
 
+    def _is_pod_restart_required(self) -> None:
+        """Checks if existing NAD config is different then the required ones which are set in the configuration  # noqa: E501, W505.
+
+        If there is difference, sets the _restart_pod attribute to True.
+        """
+        if (
+            self.get_nad_active_mtu_size(ACCESS_NETWORK_ATTACHMENT_DEFINITION_NAME)
+            != self._get_access_interface_mtu_config()  # noqa: W503
+        ):
+            self._restart_pod = True
```

Comment:
we can return after this. no need to check the second condition

Reply:
Good catch. We can add a return here.

Reply:
Done